src/test/java/org/broadinstitute/hellbender/utils/UtilsUnitTest.java

```diff
@@ -731,4 +731,153 @@ public void testGetDuplicatedItems(final Collection<?> collection, final Set<?>
+                    new Object[] { allDelimiterTestString.replace(":", "TS"),    "TS",   Arrays.asList( allDelimiterTestString.replace(":", "TS").split("TS") ) },
+                    new Object[] { frontDelimiterTestString.replace(":", "TS"),  "TS",   Arrays.asList( frontDelimiterTestString.replace(":", "TS").split("TS") ) },
+                    new Object[] { middleDelimiterTestString.replace(":", "TS"), "TS",   Arrays.asList( middleDelimiterTestString.replace(":", "TS").split("TS") ) },
+                    new Object[] { backDelimiterTestString.replace(":", "TS") ,  "TS",   Arrays.asList( backDelimiterTestString.replace(":", "TS").split("TS") ) },
+                    new Object[] { fullDelimiterTestString.replace(":", "TS") ,  "TS",   Arrays.asList( fullDelimiterTestString.replace(":", "TS").split("TS") ) }
+                )
+        );
+
+        // Create test cases for words:
+        for ( final String delim : wordsToSplitOn ) {
+            testCases.add( new Object[] { stringData, delim, Arrays.asList( stringData.split(delim) ) } );
+        }
+
+        // Create test cases for repeated substrings:
+        for ( final String delim : repeatedSubstringsToSplitOn ) {
+            testCases.add( new Object[] { stringData, delim, Arrays.asList( stringData.split(delim) ) } );
+        }
+
+        return testCases.iterator();
+    }
+
+    @DataProvider
+    private Iterator<Object[]> provideDataForTestUtilsSplitStringExhaustively() {
+
+        // Length that we want to test through:
+        final int maxStringLength = 7;
+
+        // Create single character delimiter strings:
+        final String singleCharDelimiter = "o";
+        final List<String> singleCharTestStrings = Arrays.asList("X", singleCharDelimiter);
+        final List<List<String>> exhaustiveListsForSingleChar = Utils.makePermutations( singleCharTestStrings, maxStringLength, true );
+
+        // Create multi-character delimiter strings:
+        final String multiCharDelimiter = "oz";
+        // Must include the individual characters of the multi-char delimiter here:
+        final List<String> multiCharTestStrings = Arrays.asList("X", "o", "z", multiCharDelimiter);
+        final List<List<String>> exhaustiveListsForMultiChar = Utils.makePermutations( multiCharTestStrings, maxStringLength, true );
+
+        final List<Object[]> testCases = new ArrayList<>();
+
+        // Add single-char cases:
+        for ( final List<String> testCase : exhaustiveListsForSingleChar ) {
+            testCases.add( new Object[] { String.join( "", testCase ), singleCharDelimiter, Arrays.asList( String.join( "", testCase ).split(singleCharDelimiter)) } );
+        }
+
+        // Add multi-char cases:
+        for ( final List<String> testCase : exhaustiveListsForMultiChar ) {
+            testCases.add( new Object[] { String.join( "", testCase ), multiCharDelimiter, Arrays.asList( String.join( "", testCase ).split(multiCharDelimiter)) } );
```

Comment:
Same comment as above applies to this `DataProvider` -- it would be easier to generate the expected output using `String.split()` within the `@Test` method itself.

Reply:
Fixed!